### PR TITLE
Add robot automation for running a MetaDeploy plan

### PR DESCRIPTION
**Pull request by davisagli**
This adds a Robot Framework suite that can run a specified MetaDeploy plan on a scratch org and capture screenshots.

To use:
1. Adjust variables in robot/suites/run_plan.robot
2. `cci task run robot --org enterprise`
3. Look at results in robot/output

@boakley I'm especially interested in your notes on anything that looks weird from a Robot Framework perspective, and whether you have any ideas for a less verbose way of translating text.

TBD how exactly we'll use this for integration testing.

### CODE REVIEW

**Comment by davisagli** [comment]
Note that it's currently set up to run against metadeploy-stg, so you don't actually need to be running metadeploy locally.

File: robot/suites/run_plan.robot [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+Go to Plan
+    [Arguments]  ${plan}
+    Go To  ${BASE_URL}/products/${product}/${plan}
+    ${text} =  Translate Text  Log In
+    Wait Until Page Contains  ${text}
+
+Log in to Org
+    ${text} =  Translate Text  Log In
+    Click button  ${text}
+    Capture Page Screenshot  ${OUTPUTDIR}/${LANG}/{index}.png
```
Comment (boakley) [inline robot/suites/run_plan.robot]: Does it make sense to use <n>.png as the filename. Why not use human readable names?  You could even write your own keyword to encapsulate the directory, so the above might be written as:

    Take screenshot  login.png

File: robot/suites/run_plan.robot [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    Input text  css:#username  &{ORG}[username]
+    Input password  css:#password  &{ORG}[password]
+    Click button  css:#Login
+    ${needs_oauth_approval}=  Run Keyword And Return Status    Element Should Be Visible   css:#oaapprove
+    Run Keyword If    ${needs_oauth_approval}  Click button  css:#oaapprove
+
+Maybe Run Preflight
+    ${text1} =  Translate Text  Start Pre-Install Validation
+    ${text2} =  Translate Text  Re-Run Pre-Install Validation
+    ${has_preflight} =  Run Keyword and Return Status  Wait Until Page Contains Element  //button[text()="${text1}"]|//button[text()="${text2}"]  timeout=5
```
Comment (boakley) [inline robot/suites/run_plan.robot]: Personally, I would spread this over several lines to keep the lines short and make it more readable.

```
${has_preflight} =  Run Keyword and Return Status  
...  Wait Until Page Contains Element  //button[text()="${text1}"]|//button[text()="${text2}"]  
...  timeout=5
```

File: robot/suites/run_plan.robot [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    Wait Until Page Contains  ${text}
+
+Log in to Org
+    ${text} =  Translate Text  Log In
+    Click button  ${text}
+    Capture Page Screenshot  ${OUTPUTDIR}/${LANG}/{index}.png
+    ${text} =  Translate Text  Use Custom Domain
+    Click link  ${text}
+    &{ORG} =  Get Org Info
+    Input text  css:#login-custom-domain  &{ORG}[instance_url]
```
Comment (boakley) [inline robot/suites/run_plan.robot]: I really like how your locators are short and sweet.

File: robot/suites/run_plan.robot [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+Load MetaDeploy
+    Open Browser with Language  ${BASE_URL}
+    Set Window Size  1024  1024
+    ${text} =  Translate Text  Select a Product to Install
+    Wait Until Page Contains  ${text}
+    Capture Page Screenshot  ${OUTPUTDIR}/${LANG}/{index}.png
+
+Go to Product
+    [Arguments]  ${product}
```
[the remark this answers is not in the document]
Comment (davisagli) [reply robot/suites/run_plan.robot]: Added.

File: robot/MetaDeploy.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import json
+import pathlib
+
+from robot.libraries.BuiltIn import BuiltIn
+from selenium.webdriver.chrome.options import Options
+
+
+class MetaDeploy:
+
```
[the remark this answers is not in the document]
Comment (davisagli) [reply robot/MetaDeploy.py]: Good point; added

File: cumulusci.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+tasks:
+    robot:
+        options:
+            suites: robot/suites
```
[the remark this answers is not in the document]
Comment (davisagli) [reply cumulusci.yml]: Done (though the RF docs about tasks do say they can also be arranged in suites)

File: cumulusci.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+            options:
+                outputdir: robot/output
```
[the remark this answers is not in the document]
Comment (davisagli) [reply cumulusci.yml]: Done

File: robot/MetaDeploy.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+            options.add_argument("--disable-background-timer-throttling")
+        else:
+            options.add_experimental_option(
+                "prefs", {"intl.accept_languages": self.lang}
+            )
+        options.add_argument(f"lang={self.lang}")
+        self.selenium.open_browser(url, browser="Chrome", options=options)
+
+    def translate_text(self, orig_text):
+        return self.translations[orig_text]
```
[the remark this answers is not in the document]
Comment (davisagli) [reply robot/MetaDeploy.py]: Good call; done

File: robot/suites/run_plan.robot [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+Suite Teardown  Close All Browsers
+
+*** Variables ***
+
+${BASE_URL}  https://metadeploy-stg.herokuapp.com
+${LANG}  en
+${PRODUCT}  npsp
+${PLAN}  install
+
+*** Test Cases ***
```
[the remark this answers is not in the document]
Comment (davisagli) [reply robot/suites/run_plan.robot]: There is an assertion, that it finished with a success message. But Tasks does make more sense here and I changed it. Unfortunately the VSCode syntax highlighter for robot files doesn't recognize `*** Tasks ***`

File: robot/suites/run_plan.robot (hunk 1 above)
Comment (davisagli) [reply robot/suites/run_plan.robot]: Changed this (it was also part of the request from the localization team)

Comment (davisagli) [reply robot/suites/run_plan.robot]: Copying and pasting a single line wasn't painful enough for me to create the separate keyword, but it did occur to me.

File: robot/suites/run_plan.robot (hunk 2 above)
Comment (davisagli) [reply robot/suites/run_plan.robot]: Nice, I didn't know I could do that.

File: robot/suites/run_plan.robot (hunk 3 above)
Comment (davisagli) [reply robot/suites/run_plan.robot]: Isn't it nice to write tests for something where we control the DOM?

Comment (boakley) [reply robot/suites/run_plan.robot]: Without a doubt!

**Pull request merged**